rest-server/src/templates/yarnContainerScript.mustache

```diff
@@ -28,10 +28,13 @@ BASH_XTRACEFD=13
 
 function exit_handler()
 {
-  printf "%s %s\n" \
-    "[DEBUG]" "Yarn container exit handler: EXIT signal received in yarn container, exiting ..."
-  pid=$(docker inspect --format={{{ inspectFormat }}} $docker_name) && kill -9 $pid ||\
-    printf "%s %s\n"  "[DEBUG]" "Yarn container exit handler tries to kill the container $docker_name that does not exist. It probably has already exited."
+  pid=$(docker inspect --format={{{ inspectFormat }}} $docker_name 2>/dev/null)
+  if [ $pid ]; then
+    printf "%s %s\n" \
+      "[DEBUG]" "Yarn container exit handler: EXIT signal received in yarn container, exiting ..."
```

Comment:
I think this should be moved to the beginning ?

Reply:
Yes, the more readable and trackable log messages should looks like:

[DEBUG] Yarn container exit handler: EXIT signal received in yarn container, performing clean up actions ...
[DEBUG] Trying to killing Docker container "xxx" ...
[DEBUG] $docker_name that does not exist. It probably has already exited.
or 
[DEBUG] $docker_name killed.

---
In reply to: [205736949](https://github.com/Microsoft/pai/pull/999#discussion_r205736949) [](ancestors = 205736949)

Reply:
refine the log to make it more readable and clear. done.